metrics/prometheus.go

```diff
@@ -1856,6 +1856,9 @@ func BaseContainerLabels(whiteList []string) func(container *info.ContainerInfo)
 				set[ContainerLabelPrefix+k] = v
 			}
 		}
+		for k, v := range container.Spec.Envs {
+			set[ContainerEnvPrefix+k] = v
+		}
```

Comment:
A test case would be greatly appreciated.

Reply:
Hi @iwankgb, after diving into the code, I found that the env is filtered while manage call container handler, e.g. containerd is at [here](https://github.com/google/cadvisor/blob/aa0afbe/container/containerd/handler.go#L146).

As a result, I added test case `TestPrometheusCollectorWithWhiteList` to cover `BaseContainerLabels` test. If there are any suggestions, pls let me know!

Reply:
BTW, I found the handler part already did have some test cases about whitelist envs, e.g. see docker [tests](https://github.com/google/cadvisor/blob/aa0afbee6da04e8d8a214333bf2747f0ae27f7a0/container/docker/handler_test.go#L121).

So... I think it's enough.